src/ferny/interaction_agent.py

```diff
@@ -251,6 +251,15 @@ async def communicate(self) -> None:
             try:
                 # We handle fds very carefully to avoid leaking them, even in case of exceptions
                 data, fds, _flags, _addr = recv_fds(self.ours, 4096, 10)
+                # read the rest of the request
+                while True:
+                    try:
+                        block = self.ours.recv(4096, socket.MSG_DONTWAIT)
+                        if not data:
+                            break
```

Comment:
In my experiments and the unit tests, this is the normal code path, i.e. this runs into EOF. I wonder if we should actually do blocking reads until EOF -- does this *ever* expect any further communication from the client? When I do that naïvely, `test/test_basic.py::test_connection_refused` just hangs indefinitely, so it seems there are at least *some* cases where we don't get an EOF.